### PR TITLE
SISRP-19059 Minimize User::Api Hub dependencies

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-19059

Following on https://github.com/ets-berkeley-edu/calcentral/pull/5834 , with (we hope!) one step to go.

Under most circumstances, Hub Contacts appear to incorporate Hub Affiliations. See SISRP-24928 for more information.

### CODE REVIEW

**Comment by raydavis** [comment]
I temporarily closed this PR when I started seeing Hub API breakages in my testing. That turned out to be an upstream bug, however.

File: app/models/user/student.rb [hunk 1]
```diff
@@ -20,12 +20,11 @@ def lookup_delegate_user_id
       CalnetCrosswalk::ByUid.new(user_id: @uid).lookup_delegate_user_id
     end
 
-    def legacy_user?
-      # Legacy IDs migrated to Campus Solutions have fewer than ten digits.
-      if campus_solutions_id.present?
-        campus_solutions_id.to_s.length < 10
-      elsif (student_id = lookup_student_id)
-        student_id.to_s.length < 10
+    def legacy_student?(id = nil)
```
Comment (paulkerschen) [inline app/models/user/student.rb]: A sensible rename.

File: app/models/hub_edos/user_attributes.rb [hunk 2]
```diff
@@ -26,9 +54,10 @@ def get_edo
     def get
       wrapped_result = handling_exceptions(@uid) do
         result = {}
-        if (edo = get_edo)
+        get_ids(result)
+        if @campus_solutions_id.present? && (edo = get_edo)
+          identifiers_check(edo)
```
Comment (paulkerschen) [inline app/models/hub_edos/user_attributes.rb]: @raydavis This class seems to be opting for more optional parentheses than is our convention.

**Comment by paulkerschen** [comment]
Apart from a mindless quibble about punctuation, this LGTM.

**Comment by raydavis** [comment]
So unidiomatic! I would I had bestowed that time in the tongues that I have in fencing, dancing and bear-baiting.

**Comment by paulkerschen** [comment]
Thanks @raydavis!

File: app/models/hub_edos/user_attributes.rb [hunk 3]
```diff
@@ -55,31 +84,22 @@ def has_role?(*roles)
...
-      if identifiers.present? && (cs_id_hash = identifiers.select {|id| id[:type] == 'student-id'}.first)
-        @campus_solutions_id ||= cs_id_hash[:id]
-        result[:campus_solutions_id] = @campus_solutions_id
-        result[:student_id] = @campus_solutions_id if result[:roles].slice(:student, :exStudent, :applicant).has_value?(true)
+      if identifiers.blank?
+        logger.error "No 'identifiers' found in CS attributes #{edo} for UID #{@uid}, CS ID #{@campus_solutions_id}"
       else
-        logger.error "No 'student-id' found in CS Identifiers #{identifiers} for UID #{@uid}" if identifiers.present?
-        result[:student_id] = result[:legacy_student_id]
+        edo_id = identifiers.select {|id| id[:type] == 'student-id'}.first
```
Comment (redconfetti) [inline app/models/hub_edos/user_attributes.rb]: 👍

File: app/models/user/student.rb (hunk 1 above)
Comment (redconfetti) [inline app/models/user/student.rb]: 🎩

**Pull request merged**